project/Doc.scala

```diff
@@ -7,47 +10,36 @@ import sbt.Keys._
 import sbt.File
 import scala.annotation.tailrec
 
-object Unidoc {
+object Scaladoc extends AutoPlugin {
 
-  def settings(ignoreAggregates: Seq[Project], ignoreProjects: Seq[Project]) = {
-    val withoutAggregates = ignoreAggregates.foldLeft(inAnyProject) { _ -- inAggregates(_, transitive = true, includeRoot = true) }
-    val docProjectFilter = ignoreProjects.foldLeft(withoutAggregates) { _ -- inProjects(_) }
-
-    inTask(unidoc)(Seq(
-      unidocProjectFilter in ScalaUnidoc := docProjectFilter,
-      unidocProjectFilter in JavaUnidoc := docProjectFilter,
-      apiMappings in ScalaUnidoc := (apiMappings in (Compile, doc)).value
-    ))
+  object CliOptions {
+    val scaladocDiagramsEnabled = CliOption("akka.scaladoc.diagrams", true)
+    val scaladocAutoAPI = CliOption("akka.scaladoc.autoapi", true)
   }
 
-  val genjavadocEnabled = sys.props.get("akka.genjavadoc.enabled").getOrElse("false").toBoolean
-  val (unidocSettings, javadocSettings) =
-    if (genjavadocEnabled)
-      (scalaJavaUnidocSettings, genjavadocExtraSettings ++ Seq(
-        scalacOptions in Compile += "-P:genjavadoc:fabricateParams=true",
-        unidocGenjavadocVersion in Global := "0.9"))
-    else (scalaUnidocSettings, Nil)
+  override def trigger = allRequirements
+  override def requires = plugins.JvmPlugin
 
-  lazy val scaladocDiagramsEnabled = sys.props.get("akka.scaladoc.diagrams").getOrElse("true").toBoolean
-  lazy val scaladocAutoAPI = sys.props.get("akka.scaladoc.autoapi").getOrElse("true").toBoolean
+  val validateDiagrams = settingKey[Boolean]("Validate generated scaladoc diagrams")
 
-  def scaladocSettings: Seq[sbt.Setting[_]] = {
-    scaladocSettingsNoVerificationOfDiagrams ++
-      (if (scaladocDiagramsEnabled) Seq(doc in Compile ~= scaladocVerifier) else Seq.empty)
-  }
-
-  // for projects with few (one) classes there might not be any diagrams
-  def scaladocSettingsNoVerificationOfDiagrams: Seq[sbt.Setting[_]] = {
+  override lazy val projectSettings = {
     inTask(doc)(Seq(
       scalacOptions in Compile <++= (version, baseDirectory in ThisBuild) map scaladocOptions,
-      autoAPIMappings := scaladocAutoAPI
-    ))
+      autoAPIMappings := CliOptions.scaladocAutoAPI.value
```

Comment:
This looks suspiciously like a macro evaluation, please rename the `value` method—what about `get`?